OpenRA.Mods.Common/Traits/BotModules/McvManagerBotModule.cs

```diff
@@ -204,10 +217,33 @@ void DeployMcv(IBot bot, Actor mcv, bool move)
 				return null;
 			};
 
-			var baseCenter = GetRandomBaseCenter();
+			var baseCenter = GetRandomBaseCenter(distanceToBaseIsImportant);
+
+			var bc = findPos(baseCenter, baseCenter, Info.MinBaseRadius, Info.MaxBaseRadius);
+
+			if (!bc.HasValue)
+				return null;
+
+			baseCenter = bc.Value;
+
+			var wPos = world.Map.CenterOfCell(baseCenter);
+			var newBaseRadius = new WDist(Info.MaxBaseRadius * 1024);
+
+			var actors = world.FindActorsInCircle(wPos, newBaseRadius)
+				.Where(a => !a.Disposed);
+
+			var enemies = actors.Where(a => player.Stances[a.Owner] == Stance.Enemy);
+
+			if (enemies.Count() > 0)
```

Comment:
Prefer `Any()` over `Count() > 0` (any can return early and doesn't need to count the entire enumeration).